File: darwin/debug.m
// 13 may 2016
#import "uipriv_darwin.h"

// TODO don't halt on release builds

void realbug(const char *file, const char *line, const char *func, const char *prefix, const char *format, va_list ap)
{
	NSMutableString *str;
	NSString *formatted;

	str = [NSMutableString new];
	[str appendString:[NSString stringWithFormat:@"[libui] %s:%s:%s() %s", file, line, func, prefix]];
	formatted = [[NSString alloc] initWithFormat:[NSString stringWithUTF8String:format] arguments:ap];
	[str appendString:formatted];
	[formatted release];
	NSLog(@"%@", str);
	[str release];
	__builtin_trap();
}

File: unix/debug.c
// 13 may 2016
#include "uipriv_unix.h"

// TODO don't halt on release builds

void realbug(const char *file, const char *line, const char *func, const char *prefix, const char *format, va_list ap)
{
	char *a, *b;

	a = g_strdup_printf("[libui] %s:%s:%s() %s", file, line, func, prefix);
	b = g_strdup_vprintf(format, ap);
	g_critical("%s%s", a, b);
	G_BREAKPOINT();
}
